### DISCUSSION TITLE
[bug/feature] Ability to turn off the spotlights, especially in Story Mode Robot Mode

### DISCUSSION THREAD

**Discussion by Weyverling** [chrislo27/PolyrhythmMania#57] [Ideas]
[version 2.0.2-20230512a_win64]

Request: Some way to turn spotlights off as an additional Graphics & Accessibility option

Justification (/ bug report):
When the spotlights turn on in the "phase change" for the boss my computer lags out and none of my inputs can come out correctly and I die near-instantly. I think this is just a problem with my setup and probably can't be fixed, but I'd like to have some way of circumventing this lag spike to have a better chance of completing Story Mode. I'm not sure if it comes under Reduce In-game Motion normally but I have that checked too, and Robot Mode still uses the spotlights.

Attached is me trying to show this lag spike. I don't know if it just comes off as a skill issue but every single time the spotlights hit my run has died, this is just the first time I got it on tape. I was doing better before but I cropped the video for size.
![image](https://github.com/chrislo27/PolyrhythmMania/assets/83324457/1009aff5-cabf-4881-98d7-d5dfe642ebea)
Also for some reason this game loves to use 100% of my GPU. The only things I have open right now are Firefox to write this and Polyrhythm Mania. I've made the resolution as low as I can and capped the frame rate at 30 to try and counteract the spikes but I don't think that's helped either.
https://github.com/chrislo27/PolyrhythmMania/assets/83324457/5e4ade34-922e-441c-badd-fb21104c1983

**Comment by chrislo27** [comment]
Thank you for your write-up. I will prioritize having a Graphics setting to disable spotlights in the next feature release (v2.1). 

May I ask what hardware you're running on? I have had another user discuss this but their hardware was far below minimum requirements.

I have created an issue from this discussion here: https://github.com/chrislo27/PolyrhythmMania/discussions/57#discussioncomment-6723049

**Reply by Weyverling** [reply]
![image](https://github.com/chrislo27/PolyrhythmMania/assets/83324457/82488b21-b0af-4d18-8496-b42277f99f78)
![image](https://github.com/chrislo27/PolyrhythmMania/assets/83324457/077b80a5-696d-4397-aed6-a4e25d7329f4)

Yeah I think I just didn't notice the minimum spec requirements before. My bad lol
Everything else runs fine though, it's just cranking that GPU
LMK if you need something else for specs I'll try and find it

**Reply by chrislo27** [reply]
Hmm, something is very wrong with this. I have this same hardware on a laptop and it runs perfectly fine with no GPU strain. I'm suspecting this is a driver issue but unfortunately helping with this is beyond what I can help with.

**Reply by Weyverling** [reply]
![image](https://github.com/chrislo27/PolyrhythmMania/assets/83324457/f27bb1b5-7431-4771-b497-1027f0f577b8)
Yeah I think something's just wrong on my end with that that I just haven't noticed before I'll investigate over here and see what I can do

**Reply by chrislo27** [reply]
I think the video decode being active is odd, unless you're actively watching a video or something. I remember someone else I know who has similar hardware had issues when they were in a Zoom call trying to play the game. Are you sure you have nothing else running?

**Reply by Weyverling** [reply]
![image](https://github.com/chrislo27/PolyrhythmMania/assets/83324457/8cd0c243-be71-49ef-ad86-25445829e699)
Minecraft screenshot: closed all active apps, didn't mess with background processes
Second screenshot: Only File Explorer, Snipping Tool, and Paint open - I think it's using less cos it's not full screen?
![image](https://github.com/chrislo27/PolyrhythmMania/assets/83324457/084ff0bf-996e-4b73-9c05-6a4a71aa01c7)
By resize I mean the button between minimizing and closing a tab. Just pressing that makes it spike the GPU on the loading screen. Because I took pictures while not full screen to see the task manager at the same time as the game it decided to be less intensive.

**Reply by chrislo27** [reply]
The difference between windowed and fullscreen is probably just that the rendering area gets bigger with fullscreen, I believe it uses some kind of borderless windowed mode when in "fullscreen" (this is controlled by the game engine and not me).

I also tested on my laptop with the same CPU as yours and there are no issues. Plugged in, 60 FPS, the 3D usage is around 11% and video decode is zero. Notably my driver version is a lot older (2021). I would try checking for a GPU driver update from Intel first: https://www.intel.com/content/www/us/en/support/articles/000005629/graphics.html 

Another thing to check would be if the machine is thermal throttling, or overheating. It doesn't immediately seem like the case from your graphs, but that could explain limited performance. Usually I use a program like HWiNFO64 (https://www.hwinfo.com/download/) to check temperatures.

**Comment by chrislo27** [comment]
[This feature has been released as of version 2.1.0.](https://github.com/chrislo27/PolyrhythmMania/releases/tag/v2.1.0)

**Reply by Weyverling** [reply]
Thank you so much!!
